wrappers/gtimelog_parser.py

```diff
@@ -18,65 +20,136 @@ def __init__(self, config):
                                self.settings.virtual_midnight)
         self.aliases = config.get('aliases', {})
         self.line_format = config.get('line_format', '')
+        self.dispatch_blacklist_prefixes = self.parse_dispatch_blacklist_prefixes(config)
         if self.line_format == 'categorized':
             self.line_format_str = "category: task description | comment"
             self.delimiter = " "
         else:
             self.line_format_str = "task: description | comment"
             self.delimiter = ":"
 
+    def parse_dispatch_blacklist_prefixes(self, config):
+        prefixes = config.get('dispatch_blacklist_prefixes', "").split(",")
+        # Cleanup in case of spaces
+        return [
+            prefix.strip() for prefix in prefixes
+            if prefix.strip()
+        ]
+
     def skip_entry(self, entry):
         if '**' in entry:
             return True
         if entry.strip() in ('arrive', 'arrived', 'start'):
             return True
         return False
 
+    def is_dispatch_entry(self, entry):
+        """Returns whether or not this entry is a dispatch"""
+        return '++' in entry
+
+    def is_internal_entry(self, entry):
+        """Returns whether or not this entry is internal"""
+        if not self.dispatch_blacklist_prefixes:
+            return False
+        issue, description = self.get_issue_description(entry)
+        return any([
+            issue.startswith(ref)
+            for ref in self.dispatch_blacklist_prefixes
+        ])
+
+    def get_entries_to_dispatch(self, window):
+        """Returns non internal nor dispatchable nor skippable entries"""
+        return [
+            entry
+            for start, stop, duration, tags, entry in window.all_entries()
+            if not (
+                self.skip_entry(entry)
+                or self.is_dispatch_entry(entry)
+                or self.is_internal_entry(entry)
+            )
+        ]
+
+    def get_duration_to_dispatch(self, window):
+        """Returns the time to dispatch to entries
+        i.e. total amount of dispatchable entries divided byt the quantity of
+        entries to dispatch
+        """
+        total_duration = 0
+        entries_to_dispatch = self.get_entries_to_dispatch(window)
+        total_entries_to_dispatch = len(entries_to_dispatch)
+        if not total_entries_to_dispatch:
+            return total_duration
+        for start, stop, duration, tags, entry in window.all_entries():
+            if not self.is_dispatch_entry(entry):
+                continue
+            total_duration += int(duration.total_seconds())
+        return total_duration / total_entries_to_dispatch
+
+    def get_issue_description(self, entry):
+        # remove comments
+        line = entry.split('|')[0]
```

Comment:
This is a new functionality ? Should be in the readme

Reply:
No, it's just an extraction in a method to allow reuse.